components/esp_netif/lwip/esp_netif_lwip_slip_driver.c

```diff
@@ -0,0 +1,120 @@
+
+
+
+
+#include "esp_netif.h"
+#include "esp_eth.h"
+#include "esp_eth_netif_glue.h"
+#include "esp_event.h"
+#include "esp_log.h"
+
+#include "lwip/opt.h"
+#include "lwip/sio.h"
+#include "lwip/ip.h"
+#include "lwip/ip6.h"
+#include "lwip/ip6_addr.h"
+#include "lwip/netif.h"
+
+
+#include "esp_netif_lwip_internal.h"
+#include "esp_netif_lwip_slip.h"
+
+
+static const char *TAG = "esp-netif_lwip-slip_driver";
+
+
+typedef struct {
+    esp_netif_driver_base_t base;
+    lwip_slip_ctx_t         *slip_driver;
+} esp_netif_slip_driver_t;
+
+
+
+static esp_err_t esp_slip_driver_start(lwip_slip_ctx_t* driver_impl) {
+    ESP_LOGD(TAG, "%s ctx %p", __func__, driver_impl);
+
+    esp_netif_start_slip(driver_impl);
+
+    return ESP_OK;
+}
+
+void esp_netif_lwip_slip_output(lwip_slip_ctx_t *slip_ctx, void *buffer, size_t len);
+
+esp_err_t esp_slip_driver_transmit(void *slip_driver, void *buffer, size_t len) {
+    ESP_LOGD(TAG, "%s", __func__);
+    ESP_LOG_BUFFER_HEXDUMP(TAG, buffer, len, ESP_LOG_DEBUG);
+
+    lwip_slip_ctx_t* slip_ctx = (lwip_slip_ctx_t*) slip_driver;
+
+    esp_netif_lwip_slip_output(slip_ctx, buffer, len);
+
+    return ESP_OK;
+}
+
+void esp_slip_driver_receive(esp_netif_t * esp_netif, void *buffer, size_t len) {
+    ESP_LOGD(TAG, "%s", __func__);
+    ESP_LOG_BUFFER_HEXDUMP(TAG, buffer, len, ESP_LOG_DEBUG);
+
+    esp_netif_receive(esp_netif, buffer, len, NULL);
+}
+
+static esp_err_t esp_slip_post_attach(esp_netif_t * esp_netif, void * args)
+{
+    esp_netif_slip_driver_t *driver = (esp_netif_slip_driver_t *) args;
+    driver->base.netif = esp_netif;
+
+
+    ESP_LOGD(TAG, "%s (netif: %p args: %p)", __func__, esp_netif, args);
+
+    const esp_netif_driver_ifconfig_t driver_ifconfig = {
+            .driver_free_rx_buffer = NULL,
+            .transmit = esp_slip_driver_transmit,
+            .handle = driver->slip_driver,
+    };
+
+    driver->base.netif = esp_netif;
+    ESP_ERROR_CHECK(esp_netif_set_driver_config(esp_netif, &driver_ifconfig));
+    
+    esp_slip_driver_start(driver->slip_driver);
+    
+    return ESP_OK;
+}
+
+esp_err_t esp_slip_driver_netif_set_default_handlers(esp_netif_slip_driver_t *driver, esp_netif_t * esp_netif)
+{
+    ESP_LOGD(TAG, "%s", __func__);
+
+    // TODO: do we need to attach any handlers here?
```

Comment:
Yes, we typically create handlers for start/stop actions to set the corresponding netif up/down. For that we also need someone to post into that event id, could be the driver code after setting up the UART.

Reply:
Alrighty, so instead of calling the slip start / stop functions here:

https://github.com/espressif/esp-idf/pull/4985/files#diff-a02d6b6299ad381db96e4ff380108574R163

I should emit start/stop events and let the default handlers manage em?